multidim_threshold/projection.py

```diff
@@ -54,6 +61,21 @@ def generate_proj_vecs(lo, hi, direc=None):
         yield from vecs
         vecs = [ProjVec(r, direc) for r in next_roots(lo, hi, vecs)]
 
+def generate_axes_intersects(lo, hi, member_oracles, *, searches=None):
```

Comment:
So I typically reserve the "generate" name for generators.
I think `axes_intersections` is descriptive enough.